src/stac_utils/google.py

```diff
@@ -1,16 +1,20 @@
 import json
 import os
 import sys
-
-from inflection import parameterize, underscore
 from typing import Any, List, Union, Mapping, Sequence
 
 from google.cloud import storage, bigquery
 from google.oauth2 import service_account
 from googleapiclient.discovery import build
+from google.api_core.retry import if_exception_type, Retry
+from google.api_core.exceptions import InternalServerError
+from inflection import parameterize, underscore
+
 
 from .listify import listify
 
+RETRY_EXCEPTIONS = [InternalServerError]
```

Comment:
Is there a way to make this dynamic? i.e. If there are certain errors that we see and want to trigger a retry on in some cases but not all, is there a way to override this? It may not be worth spending a ton of time on right now if it's not easy, but it's definitely something I could see come up.

Reply:
Internal Server Error is the ephemeral 500 error that we get that doesn't have anything specific. There are other errors for other types of job failures that we can add if we like but that was the one I saw examples for.

Reply:
I don't know of any off-hand. This is just something that I can very easily imagine that we'd find other exceptions to handle the same way in the future. But if another sort of error came up, we may want to handle it with a retry in some jobs but not all.

Reply:
If it's hard to make it modular, we can deal with that when it comes up. But if it's easy,  it's worth doing.